cmd/sync-tags/main.go

```diff
@@ -199,13 +206,19 @@ func main() {
 			continue
 		}
 
-		// skip if it already exists in origin
-		if _, found := bTagCommits[bName]; found {
-			continue
-		}
+		// check if the tags exist either at origin or locally
+		_, nonSemverTagAtOrigin := bTagCommits[bName]
+		nonSemverTagExists := nonSemverTagAtOrigin || tagExists(r, bName)
+
+		_, semverTagAtOrigin := bTagCommits[semverTag]
+		semverTagExists := semverTagAtOrigin || tagExists(r, semverTag)
 
-		// do not override tags (we build master first, i.e. the x.y.z-alpha.0 tag on master will not be created for feature branches)
-		if tagExists(r, bName) {
+		// skip tags if they exist locally or at origin already
+		// we want to ensure to not override tags
+		// (we build master first, i.e. the x.y.z-alpha.0 tag on master will not be created for feature branches)
+		if *publishSemverTags && semverTagExists && nonSemverTagExists {
+			continue
+		} else if nonSemverTagExists {
```

Comment:
Updated the check for tags at origin and locally. Verified this and pushed the latest results to the test repo, but can you sanity check this again?